### PR TITLE
use lua-posix to fork and set the environment in the pun_pre_hook

### LINKED ISSUE #1078: Change the prehook to use fork+exec so we can pass options via ENV

### PR DESCRIPTION
**Pull request by johrstrom**
Fixes #1078

Setting to draft because I'd like to get #1090 in first. 

I'd also like to get in some sort of failsafe in. Whatever we fork+exec should have some sort of guardrail in place so that after so much time if it hasn't executed, we force kill it. So clearly a test case to handle this would be good.

### CODE REVIEW

**Comment by treydock** [comment]
Added doc label, because this will change the install instructions for OnDemand. We do not have direct dependency on EPEL, and EPEL will only be needed for EL7. For EL8 the lua-posix module comes from `powertools` repo which is part of OS but not enabled by default on minimal install but very common to have enabled.

File: Dockerfile.dev [hunk 1]
```diff
@@ -5,8 +5,9 @@ ARG UID=1000
 ARG GID=1000
 
 RUN dnf reinstall -y httpd-tools && \
+    dnf install -y epel-release && \
```
Comment (treydock) [inline Dockerfile.dev:8]: EPEL on EL8 is not required to access `lua-posix`.  It comes from powertools repo.

**Comment by ericfranz** [comment]
After renaming `fork_and_exec` to something more appropriate, perhaps `capture2e_with_env`, this LGTM. This is only called if pre_hook_root_cmd exists, and we don't need to advertise this yet, or we could do so as "experimental". We will use it in production ourselves for a while before we remove that label and then we can be confident in the solution.

File: spec/fixtures/hooks/good_pre_hook [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 echo "input args are $@" > /tmp/hook.out 2>&1
 
-if [[ -n "$OIDC_ACCESS_TOKEN" ]]; then
+if [[ -n "$OOD_OIDC_ACCESS_TOKEN" ]]; then
```
Comment (johrstrom) [inline spec/fixtures/hooks/good_pre_hook:5]: I had to rebase to pick up the tests to modify this bit. Just FYI when we go to document that all the env variables will be prefixed with `OOD_`.

**Pull request merged**